google/cloud/internal/async_connection_ready.cc

```diff
@@ -27,7 +27,8 @@ AsyncConnectionReadyFuture::AsyncConnectionReadyFuture(
     : cq_(std::move(cq)), channel_(std::move(channel)), deadline_(deadline) {}
 
 future<Status> AsyncConnectionReadyFuture::Start() {
-  RunIteration(channel_->GetState(true));
+  channel_->GetState(true);
```

Comment:
nit: I think this call is superfluous

Reply:
I looked at that too, but it causes the channel to try to make a connection if it is currently IDLE.

Reply:
won't it get called here anyways? https://github.com/googleapis/google-cloud-cpp/blob/55f718928d4ff14515dde192763715c89a7746bf/google/cloud/internal/async_connection_ready.cc#L74

Reply:
Removed